wagtail/contrib/redirects/middleware.py

```diff
@@ -17,11 +19,32 @@ def _get_redirect(request, path):
     site = Site.find_for_request(request)
     try:
         return models.Redirect.get_for_site(site).get(old_path=path)
+    except models.Redirect.DoesNotExist:
+        # Check for wildcard redirects
+        wildcard_redirects = models.Redirect.get_for_site(site).filter(
+            old_path__endswith="*"
+        )
+
+        for redirect in wildcard_redirects:
+            wildcard_path = redirect.old_path.rstrip("*")
+            wildcard_redirects_prefix = models.Redirect.get_for_site(site).filter(
+                old_path__startswith=wildcard_path
+            )
+            redirect_paths = [r.old_path for r in wildcard_redirects_prefix]
+            matched_paths = []
+
+            for rp in redirect_paths:
+                rp = os.path.normpath(rp)
+                matched_paths.extend(glob.glob(os.path.join(rp + "*"), recursive=True))
```

Comment:
`glob.glob` hits the filesystem, which we don't want. You might need to use `fnmatch.fnmatch` instead.